src/layouts/default.vue

```diff
@@ -86,54 +87,54 @@ const embeddedPage = {
     VFooter,
     VModalTarget,
     VTeleportTarget,
-    VSidebarTarget,
     VGlobalAudioSection,
   },
   layout: 'embedded',
   head() {
     return this.$nuxtI18nHead({ addSeoAttributes: true, addDirAttribute: true })
   },
   setup() {
+    const { app } = useContext()
+
+    const uiStore = useUiStore()
     const featureFlagStore = useFeatureFlagStore()
+    const searchStore = useSearchStore()
+
     const isNewHeaderEnabled = computed(() =>
       featureFlagStore.isOn('new_header')
     )
+    const { isDesktopLayout: isDesktop, updateLayout } = useLayout({ app })
+
+    /**
+     * If cookies state does not match the state on mounted
+     * (e.g., if the cookies were saved when the screen was `sm`,
+     * and then a page is opened on SSR on a `lg` screen),
+     * we update the cookie state.
+     */
+    onMounted(() => {
+      updateLayout(isDesktop.value)
+    })
```

Comment:
Could/should this be moved into the `useLayout` composable itself? I don't have a super strong inclination either way other than moving it into the composable would maybe localise the full functionality of the composable whereas this is a little more spread out. It also requires duplicating this in all the root layouts, which could be easy to accidentally forget.

Reply:
I've learned with the filters visibility composable, if you have `onMounted` inside a composable, it runs whenever any component that uses it is mounted. While `useLayout` is only used in the two `layout` components, I still feel that it's easier to see when the `onMounted` runs.
While we can accidentally forget to add `onMounted` to the root layout, it might be more difficult to debug if we accidentally import `useLayout` into a non-root layout file and then it runs `onMounted` when we don't need it.